api/v1alpha3/webhooks.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func aggregateObjErrors(obj runtime.Object, allErrs field.ErrorList) error {
```

Comment:
If you change obj's type to `metav1.Object` you should be able to eliminate the `meta.Accessor` call and error handling. You'd then change all the calls to this function to pass in `r.ObjectMeta` instead of `r` as the obj.

Reply:
Can't get the kind of the object from metav1.Object though?

Reply:
Good point